src/adapter.ts

```diff
@@ -71,19 +71,21 @@ export class CppUTestAdapter implements TestAdapter {
 	public async run(tests: string[]): Promise<void> {
 		this.testStatesEmitter.fire(<TestRunStartedEvent>{ type: 'started', tests });
 		this.log.info('Running tests');
-		await this.updateTests();
-		if (tests.length == 1 && tests[0] == this.mainSuite.id) {
-			await this.root.RunAllTests();
-		} else {
-			await this.root.RunTest(...tests);
+		if (await this.updateTests())
```

Comment:
Nicely done. If you want you could make it more "promisey" e.g. in the updateTests() throw an error to propagate it up and in the calling methods do something like:
```js
this.updateTests
.then(success => {
...
})
.catch(err => this.testsEmitter...)
```
But I am fine with it as it is

Reply:
Any reason to use `.then().catch()` instead of ordinary try/catch blocks? All the callers of `updateTests()` are already async methods, doesn't that result in the same behavior under the hood? I am not as fluent in TypeScript's async/Promises as I am in, e.g., Python's, but I thought those were equivalent ways of saying the same thing.

Reply:
Technically it is almost the same. It is more a style thing, but I was just perusing the rest of the code and I remember that I wanted to make it more functional but didn't go through all of it anyway. So it's fine as it is. I will merge it and maybe refactor "later" :)